admin/app/routes/authenticated/sessions/session.js

```diff
@@ -5,6 +5,11 @@ import { inject as service } from '@ember/service';
 export default class SessionRoute extends Route {
   @service notifications
 
+  setupController(controller, model) {
+    super.setupController(controller, model);
+    this.controllerFor('authenticated.sessions.session').inputId = model.id;
+  }
+
```

Comment:
Pas besoin d'écrire ça ici je pense, à ta place je ferais plutôt la modif dans le constructeur du controller :
```js
export default class SessionController extends Controller {
  @service router;
  @tracked inputId;

   constructor() {
     super(...arguments);
     this.inputId = this.model.id; // ça devrait mettre à jour le champ input puisque la variable est traquée.
   }
   /* ...reste */
}
```

Reply:
Bon malgré quelques tests différents je n'arrive pas à accéder au model depuis le controller sans passer par "setupController" ... J'ai essayé en mettant la fonction "model" dans la route explicitement et en rajouter un "alias" dans le controller mais ni l'un ni l'autre (même combiné) me permet d'accéder au model dans le controller ...

Reply:
> The model is passed from the Route to the Controller by default using the setupController() function.
(https://guides.emberjs.com/v3.17.0/routing/controllers/)
Je crois pas qu'on puisse faire autrement @laura-bergoens, en tout cas j'ai pas trouvé en regardant la doc de l'API et les tutos sur les controllers

Reply:
Ok, pour que l'explication soit gravée dans l'histoire :
La proposition que j'ai faite au-dessus est, à mon sens, la plus logique. Il s'agit du controller de la route sessions/session/:session_id, ce qui veut dire que logiquement on arrive dans le controller avec un modèle tout prêt retourné par l'objet route correspondant. 
En essayant de le mettre en place, on tombe sur une erreur toute bizarre, `this.model is undefined`.
On debug, et on s'aperçoit que quand on se rend sur sessions/list, le hook d'initialisation du controller pour la route sessions/session/:id est AUSSI appelé (bizarre car ce n'est pas sa route).
On apprend en fait que c'est à cause du helper link-to. en effet, dans la liste des sessions, on a mis des link-to pour avoir un lien vers la page de détails d'une session. Et en fait l'helper link-to provoque le init hook du controller de sessions/session/:id, qui n'a à ce moment là pas encore de modèle. Apparemment c'est un comportement connu et attendu du helper, j'ai pas bien compris, pourquoi, une histoire de query params. Bref, j'ai pas trouvé de contournement, donc je n'ai pas de contre-proposition.
https://github.com/emberjs/ember.js/issues/15973

Reply:
@laura-bergoens ce qu'il y a aussi c'est que le constructeur du `Controller` c'est vraiment pas un endroit où tu veux être parce que dans Ember, [les contrôleurs sont des singletons](https://www.balinterdi.com/blog/ember-gotcha-controllers-are-singletons/) !

Reply:
@Krysthalia En fait plus je regarde les internets, plus ta solution me paraît être la meilleure tout compte fait. All good pour moi :+1: